xbmc/cores/VideoPlayer/DVDCodecs/Video/MMALCodec.cpp

```diff
@@ -849,6 +855,18 @@ bool CMMALVideo::GetPicture(DVDVideoPicture* pDvdVideoPicture)
     pDvdVideoPicture->iFlags  = DVP_FLAG_ALLOCATED;
     if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_FLAG_USER3)
       pDvdVideoPicture->iFlags |= DVP_FLAG_DROPPED;
+
+    if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_VIDEO_FLAG_INTERLACED)
+    {
+      pDvdVideoPicture->iFlags |= DVP_FLAG_INTERLACED;
+      if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_VIDEO_FLAG_TOP_FIELD_FIRST)
+        pDvdVideoPicture->iFlags |= DVP_FLAG_TOP_FIELD_FIRST;
+      if (!firstInterlacedFrame)
+      {
+        pDvdVideoPicture->dts = DVD_NOPTS_VALUE;
+        pDvdVideoPicture->pts = DVD_NOPTS_VALUE;
```

Comment:
VideoPlayerVideo can only guess while decoder should know. Decoder gets 50 distinct access units with pts. Eating half of the pts values reduces information.
VideoPlayerVideo does not know that decoder combines two fields into a frame with only one pts. It has some fallback logic to set pts but this can't ever be as accurate than what decoder can do. Decoder has more information and is in charge of setting pts.

> Are you suggesting using the framerate from hints?

Definitely not. That would be worse than guessing.